agr_java_core/src/main/java/org/alliancegenome/core/translators/document/PhenotypeTranslator.java

```diff
@@ -0,0 +1,281 @@
+                        doclet.setUrl(doTerm.getMgiLink());
+                        doclet.setDiseaseUrl(doTerm.getMgiLink());
+                    }
+                    if (speciesType == SpeciesType.ZEBRAFISH && doTerm.getZfinLink() != null) {
+                        doclet.setUrl(doTerm.getZfinLink());
+                        doclet.setDiseaseUrl(doTerm.getZfinLink());
+                    }
+                    if (speciesType == SpeciesType.HUMAN && doTerm.getHumanLink() != null) {
+                        doclet.setUrl(doTerm.getHumanLink());
+                    }
+                    if (speciesType == SpeciesType.WORM && doTerm.getWormbaseLink() != null) {
+                        doclet.setUrl(doTerm.getWormbaseLink());
+                        doclet.setDiseaseUrl(doTerm.getWormbaseLink());
+                    }
+                    return doclet;
+                })
+                .collect(toList());
+
+        return sourceDoclets;
+    }
+
+    @Override
+    protected Phenotype documentToEntity(PhenotypeDocument document, int translationDepth) {
+        return null;
+    }
+
+    private List<PublicationDoclet> getPublicationDoclets(List<PhenotypeEntityJoin> phenotypeEntityJoins) {
+        Set<PublicationDoclet> publicationDocuments = phenotypeEntityJoins.stream()
+                // filter out records that do not have valid pub / evidence code entries
+                .filter(phenotypeEntityJoin ->
+                        getPublicationDoclet(phenotypeEntityJoin, phenotypeEntityJoin.getPublication()) != null
+                )
+                .map(diseaseGeneJoin -> {
+                    Publication publication = diseaseGeneJoin.getPublication();
+                    return getPublicationDoclet(diseaseGeneJoin, publication);
+                })
+                .collect(Collectors.toSet());
+        List<PublicationDoclet> pubDocletListRaw = new ArrayList<>(publicationDocuments);
+        pubDocletListRaw.sort(PublicationDoclet::compareTo);
+
+        // get evidence codes for same pub onto s
+        List<PublicationDoclet> pubDocletList = new ArrayList<>();
+        for (PublicationDoclet doclet : pubDocletListRaw) {
+            PublicationDoclet existingDoclet = null;
+            for (PublicationDoclet finalDoclet : pubDocletList) {
+                if (doclet.compareTo(finalDoclet) == 0) {
+                    existingDoclet = finalDoclet;
+                }
+            }
+            if (existingDoclet == null) {
+                pubDocletList.add(doclet);
+            } else {
+//                existingDoclet.getEvidenceCodes().addAll(doclet.getEvidenceCodes());
+            }
+        }
+        return pubDocletList;
+    }
+
+    private SpeciesDoclet getSpeciesDoclet(Gene gene) {
+        return gene.getSpecies().getType().getDoclet();
+    }
+
+    public Iterable<PhenotypeAnnotationDocument> translateAnnotationEntities(List<Phenotype> phenotypeList) {
```

Comment:
Can translateAnnotationEntities for PhenotypeAnnotationDocument be turned into its own translator so this class won't be so large?